server/services/omnichannel-voip/service.ts

```diff
@@ -52,24 +117,115 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
 			_.pluck(allocatedExtensions, 'extension'),
 		) as string[];
 		this.logger.debug({ msg: 'getAvailableExtensions()', found: filtered.length });
-		return {
-			result: filtered,
-		};
+		return filtered;
 	}
 
-	async getExtensionAllocationDetails(): Promise<IOmnichannelVoipServiceResult> {
+	async getExtensionAllocationDetails(): Promise<IAgentExtensionMap[]> {
 		const allocatedExtensions = await this.getAllocatedExtesionAllocationData({
 			username: 1,
 			roles: 1,
 			extension: 1,
 		});
 		this.logger.debug({ msg: 'getExtensionAllocationDetails() all extension length ', length: allocatedExtensions.length });
+		return allocatedExtensions.map((user: any) => ({
+			_id: user._id,
+			agentName: user.username,
+			extension: user.extension,
+		}));
+	}
+
+	/* Voip calls */
+	async findAgent(agentId: string): Promise<any> {
+		return this.normalizeAgent(agentId);
+	}
+
+	async getNewRoom(
+		guest: ILivechatVisitor,
+		agent: any,
+		rid: string,
+		roomInfo: any,
```

Comment:
Replace `any` with the actual type of those arguments.

Reply:
Fixed.